components/backup/src/writer.rs

```diff
@@ -65,17 +75,42 @@ impl BackupWriter {
                         // it, we need to add the prefix manually.
                         let data_key_default = keys::data_key(&default.0);
                         self.default.put(&data_key_default, &default.1)?;
-                        self.default_written = true;
+                        value_in_default.replace(true);
                     }
                     assert!(!write.0.is_empty());
                     let data_key_write = keys::data_key(&write.0);
                     self.write.put(&data_key_write, &write.1)?;
-                    self.write_written = true;
                 }
                 TxnEntry::Prewrite { .. } => {
-                    return Err(Error::Other("prewrite is not supported".into()))
+                    value_in_default.take();
+                    return Err(Error::Other("prewrite is not supported".into()));
                 }
             }
+            match value_in_default {
```

Comment:
How about checking if need_checksum first?

Reply:
we need to update total_kvs even `need_checksum = false` 😓